[BUG] Get workout test failed

## 문제 상황
- [x] (https://github.com/depromeet/pumping-server/actions/runs/5429785514)

## 서버 로그
```
> Task :test

WorkoutServiceTest > 각_크루별_가입일_이후_최대_7개_운동_데이터만_조회한다() FAILED
    org.opentest4j.AssertionFailedError at WorkoutServiceTest.kt:101

25 tests completed, 1 failed

> Task :test FAILED

FAILURE: Build failed with an exception.
```

## 문제 원인 및 해결과정
+ workout 컬렉션 ID 값이 하드 코딩 되어 있어서 DB에 계속 하나만 저장이 돼서 테스트 실패

--- a/pumping/src/main/kotlin/com/dpm/pumping/workout/application/WorkoutService.kt
+++ b/pumping/src/main/kotlin/com/dpm/pumping/workout/application/WorkoutService.kt
@@ -27,7 +27,10 @@ class WorkoutService(
     fun createWorkout(
         request: WorkoutCreateDto.Request, user: User
     ): WorkoutCreateDto.Response {
-        val workout = Workout.of(user.uid!!, request.currentCrew, request.timers)
+        val crew = user.currentCrew
+            ?: throw IllegalArgumentException("아직 크루에 참여하지 않아 운동 기록을 저장할 수 없습니다.")
+
+        val workout = Workout.of(user.uid!!, crew.crewId!!, request.timers)
         val created = workoutRepository.save(workout)
         return WorkoutCreateDto.Response(created.workoutId!!)
     }

--- a/pumping/src/main/kotlin/com/dpm/pumping/workout/application/WorkoutService.kt
+++ b/pumping/src/main/kotlin/com/dpm/pumping/workout/application/WorkoutService.kt
@@ -29,7 +29,7 @@ class WorkoutService(
     ): WorkoutCreateDto.Response {
         val workout = Workout.of(user.uid!!, request.currentCrew, request.timers)
         val created = workoutRepository.save(workout)
-        return WorkoutCreateDto.Response(created.workoutId)
+        return WorkoutCreateDto.Response(created.workoutId!!)
     }
 
     fun getWorkouts(userId: String): WorkoutGetDto.Response {

--- a/pumping/src/main/kotlin/com/dpm/pumping/workout/domain/entity/Workout.kt
+++ b/pumping/src/main/kotlin/com/dpm/pumping/workout/domain/entity/Workout.kt
@@ -12,7 +12,7 @@ import java.util.*
 @Document(collection = "workout")
 data class Workout(
     @Id
-    var workoutId: String,
+    var workoutId: String? = null,
     var userId: String,
     val currentCrew: String,
     var timers: List<Timer> = listOf(),

--- a/pumping/src/main/kotlin/com/dpm/pumping/workout/dto/WorkoutCreateDto.kt
+++ b/pumping/src/main/kotlin/com/dpm/pumping/workout/dto/WorkoutCreateDto.kt
@@ -4,7 +4,6 @@ package com.dpm.pumping.workout.dto
 class WorkoutCreateDto {
 
     data class Request(
-        val currentCrew: String,
         val timers: List<TimerDto>
     )
 

--- a/pumping/src/test/kotlin/com/dpm/pumping/workout/application/WorkoutServiceTest.kt
+++ b/pumping/src/test/kotlin/com/dpm/pumping/workout/application/WorkoutServiceTest.kt
@@ -57,10 +57,9 @@ class WorkoutServiceTest(
         )
 
         val request = WorkoutCreateDto.Request(
-            currentCrew = "crew01",
             timers = listOf(timerDto)
         )
-        val testUser = createUser(currentCrew = null)
+        val testUser = createUser(currentCrew = createCrew("crewO1"))
 
         val response = workoutService.createWorkout(request, testUser)
 

--- a/pumping/src/test/kotlin/com/dpm/pumping/workout/application/WorkoutServiceTest.kt
+++ b/pumping/src/test/kotlin/com/dpm/pumping/workout/application/WorkoutServiceTest.kt
@@ -79,11 +79,11 @@ class WorkoutServiceTest(
         }.hasMessageContaining("아직 크루에 참여하지 않아 운동 기록이 존재하지 않습니다.")
     }
 
-    // @Test
+     @Test
     fun 각_크루별_가입일_이후_최대_7개_운동_데이터만_조회한다() {
         // given
-        val crew1 = createCrew()
-        val crew2 = createCrew()
+        val crew1 = createCrew("crew01")
+        val crew2 = createCrew("crew02")
         val testUser = createUser(currentCrew = crew1)
         val timer = createTimer(WorkoutPart.ARM)
         val crew1FirstDayWorkout = createWorkout(listOf(timer), "2023-06-22T10:00:00", crew1.crewId!!, testUser)
@@ -99,14 +99,14 @@ class WorkoutServiceTest(
         // then
         val result = response.workouts!!.toList()
         assertThat(result.size).isEqualTo(2)
-        assertThat(result[0].workoutDate).isEqualTo(crew1FirstDayWorkout.createDate)
-        assertThat(result[1].workoutDate).isEqualTo(crew1SecondDayWorkout.createDate)
+        assertThat(result[0].workoutDate).isEqualTo(crew1FirstDayWorkout.createDate.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME))
+        assertThat(result[1].workoutDate).isEqualTo(crew1SecondDayWorkout.createDate.format(DateTimeFormatter.ISO_LOCAL_DATE_TIME))
     }
 
     @Test
     fun 하루치_운동_데이터의_누적_값을_반환한다() {
         // given
-        val crew = createCrew()
+        val crew = createCrew("crew01")
         val testUser = createUser(currentCrew = crew)
         val timer1 = createTimer(WorkoutPart.ARM)
         val timer2 = createTimer(WorkoutPart.HIP)
@@ -128,7 +128,7 @@ class WorkoutServiceTest(
     @Test
     fun 하루_동안_가장_많이_운동한_부위를_반환한다() {
         // given
-        val crew = createCrew()
+        val crew = createCrew("crew01")
         val testUser = createUser(currentCrew = crew)
         val timer1 = createTimer(WorkoutPart.ARM)
         val timer2 = createTimer(WorkoutPart.CHEST)
@@ -159,7 +159,6 @@ class WorkoutServiceTest(
 
     private fun createWorkout(timers: List<Timer>, createDate: String, crewId: String, user: User): Workout {
         return Workout(
-            workoutId = "workout01",
             userId = user.uid!!,
             timers = timers,
             createDate = LocalDateTime.parse(createDate, DateTimeFormatter.ISO_LOCAL_DATE_TIME),
@@ -167,9 +166,9 @@ class WorkoutServiceTest(
         )
     }
 
-    private fun createCrew(): Crew {
+    private fun createCrew(id: String): Crew {
         return Crew(
-            crewId = "crew01",
+            crewId = id,
             crewName = "name",
             code = "0248291",
             createDate = "2023-06-22T10:00:00",

--- a/pumping/src/test/kotlin/com/dpm/pumping/workout/presentation/WorkoutControllerTest.kt
+++ b/pumping/src/test/kotlin/com/dpm/pumping/workout/presentation/WorkoutControllerTest.kt
@@ -71,7 +71,6 @@ class WorkoutControllerTest(
     @Test
     fun createWorkout() {
         val request = Request(
-            currentCrew = "crew01",
             timers = listOf(
                 TimerDto(
                     time = 600,
@@ -104,7 +103,6 @@ class WorkoutControllerTest(
                     preprocessRequest(prettyPrint()),
                     preprocessResponse(prettyPrint()),
                     requestFields(
-                        fieldWithPath("currentCrew").type(JsonFieldType.STRING).description("현재 참여중인 크루 아이디"),
                         fieldWithPath("timers[].time").type(JsonFieldType.NUMBER).description("운동 시간"),
                         fieldWithPath("timers[].heartbeat").type(JsonFieldType.NUMBER).description("심박수"),
                         fieldWithPath("timers[].calories").type(JsonFieldType.NUMBER).description("소모 칼로리"),

--- a/pumping/src/test/kotlin/com/dpm/pumping/workout/repository/WorkoutRepositoryTest.kt
+++ b/pumping/src/test/kotlin/com/dpm/pumping/workout/repository/WorkoutRepositoryTest.kt
@@ -39,7 +39,7 @@ class WorkoutRepositoryTest(
         )
 
         val savedRecord = workoutRepository.save(workout)
-        val foundRecord = workoutRepository.findById(savedRecord.workoutId)
+        val foundRecord = workoutRepository.findById(savedRecord.workoutId!!)
 
         assertThat(foundRecord.isPresent).isTrue
         assertThat(foundRecord.get()).isEqualTo(savedRecord)

